### PR TITLE
Ref/feature/no 3d scrolling context menu

### PR DESCRIPTION
**Pull request by KingOfBerries**
The 3D view check now also checks if the mouse is over the context menu and if so doesn't allow 3d view input.
Also renamed the private fields to match naming convention

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-37q9g4y Don't zoom scene, when you scroll while hovering over a context menu](https://app.clickup.com/t/37q9g4y)

File: Assets/Scripts/Interaction/GeneralInputInteraction.cs [hunk 1]
```diff
@@ -41,30 +42,32 @@ public class GeneralInputInteraction : MonoBehaviour
...
-            _workspaceSaveSystem = workspaceSaveSystem;
-            _gizmoState = gizmoState;
-            _unityState = unityState;
-            _inputHelper = inputHelper;
-            _sceneDirectoryState = sceneDirectoryState;
-            _cameraState = cameraState;
-            _editorEvents = editorEvents;
-            _typeScriptGenerationSystem = typeScriptGenerationSystem;
-            _entitySelectSystem = entitySelectSystem;
+            this.sceneSaveSystem = this.sceneSaveSystem;
```
Comment (cblech) [inline Assets/Scripts/Interaction/GeneralInputInteraction.cs:48]: The second `this.` should not be there

Comment (cblech) [inline Assets/Scripts/Interaction/GeneralInputInteraction.cs]: The second `this.` should not be there

**Review by cblech — CHANGES_REQUESTED** [review]

**Comment by KingOfBerries** [comment]
That would have been troublesome if left unnoticed, sorry. Should be fixed now

**Review by cblech — APPROVED** [review]

**Pull request merged**